backend/clubs/views.py

```diff
@@ -4861,6 +4863,39 @@ def get_queryset(self):
         )
 
 
+class WhartonCyclesView(viewsets.ModelViewSet):
+    """
+    update: Update application cycle
+    list: Get list of all application cycles
+    """
+
+    permission_classes = [WhartonApplicationPermission | IsSuperuser]
+    http_method_names = ["get", "post", "put", "patch", "delete"]
+    serializer_class = ApplicationCycleSerializer
+
+    def get_queryset(self):
+        return ApplicationCycle.objects.all().order_by("end_date")
+
+    def list(self, request, *args, **kwargs):
+        return super().list(request, *args, **kwargs)
+
+    def update(self, *args, **kwargs):
+        """
+        Updates times for all applications with cycle
+        """
+        applications = ClubApplication.objects.filter(
+            application_cycle=self.get_object()
+        )
+        for app in applications:
+            app.application_start_time = self.get_object().start_date
+            app.application_end_time = self.get_object().end_date
+            if app.result_release_time < app.application_end_time:
+                app.result_release_time = app.application_end_time
+                +datetime.timedelta(days=10)
+            app.save()
```

Comment:
This won't affect prod performance but it would be good practice to use `bulk_update`!